src/freenet/client/filter/GenericReadFilterCallback.java

```diff
@@ -274,7 +260,49 @@ public String processURI(String u, String overrideType, boolean forBaseHref, boo
 			throw new CommentException(l10n("protocolNotEscaped", "protocol", uri.getScheme()));
 		}
 	}
-	
+
+	@Override
+	public String processURI(String u, String overrideType, String forceHostPort)
+			throws CommentException {
+		URI uri;
+		String filtered;
+		try {
+			filtered = processURI(makeURIAbsolute(u), overrideType, true, false);
+			uri = URIPreEncoder.encodeURI(filtered).normalize();
+		} catch (URISyntaxException e1) {
+			if(logMINOR) Logger.minor(this, "Failed to parse URI: "+e1);
+			throw new CommentException(l10n("couldNotParseURIWithError", "error", e1.getMessage()));
+		}
+		if (uri.getHost() == null) {
+			String scheme = uri.getScheme() != null ? uri.getScheme() : "http";
```

Comment:
You know, this might not be enough… we would need to get the protocol from the toadlet container as well, including using the X-Forwarded-Proto HTTP header. (Use case: I have freenet running behind a reverse proxy that does SSL for me so fred doesn’t know anything about that but the proxy sets a header fred can use to reconstruct valid URLs.)

Reply:
I hope that this is implemented correctly now.